### PR TITLE
Pal 289 data service helm charts

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: charts/data-service/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/_helpers.tpl]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/data-service/templates/autoscaler.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/autoscaler.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/data-service/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/deployment.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/data-service/templates/ingress-route.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/ingress-route.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/data-service/templates/pathparam-regex.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/pathparam-regex.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/data-service/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/data-service/templates/service.yaml:1]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/metrics-server/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/metrics-server/Chart.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/metrics-server/templates/api-service.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/metrics-server/templates/api-service.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/metrics-server/templates/deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/metrics-server/templates/deployment.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/metrics-server/templates/service-account.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/metrics-server/templates/service-account.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: charts/metrics-server/templates/service.yaml [hunk 11]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/metrics-server/templates/service.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

File: charts/data-service/templates/pv-claim.yaml [hunk 12]
```diff
@@ -0,0 +1,30 @@
+{{- if and .Values.global.persistence.enabled }}
```
Comment (ac74475) [inline charts/data-service/templates/pv-claim.yaml]: copyright and licence

File: charts/metrics-server/templates/api-service.yaml [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  labels:
+    app.kubernetes.io/name: metrics-server
+    app.kubernetes.io/part-of: metrics-server
+spec:
+  service:
+    name: metrics-server
+    namespace: kube-system
+  group: metrics.k8s.io
+  version: v1beta1
+  insecureSkipTLSVerify: true
```
Comment (ac74475) [inline charts/metrics-server/templates/api-service.yaml:28]: Does this want to be insecure, given I have seen references to having CA certs. Am not too concerned with the metrics being secure, but if all the stuff is in place to make it secure then we might as well.

Comment (nw1984) [reply charts/metrics-server/templates/api-service.yaml:28]: The metrics server is generally configured to use insecureTLS. There are various issues around enabling TLS on the metrics server e.g: https://github.com/kubernetes-sigs/metrics-server/issues/25.
I'd like to discuss this point with Alex on his return to see whether this was something specific that he saw an issue with or it was simply a cut and paste from the standard format.

File: Jenkinsfile [hunk 14]
```diff
@@ -134,7 +132,8 @@ spec:
             container('maven') {
                 configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
                     if (("${env.BRANCH_NAME}" == "develop") ||
-                            ("${env.BRANCH_NAME}" == "master")) {
+                            ("${env.BRANCH_NAME}" == "master") ||
+                            ("${env.BRANCH_NAME}" == "PAL-289-data-service-helm-charts")) {
```
Comment (developer01189998819991197253) [inline Jenkinsfile]: revert back to develops jenkinsfile

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: charts/metrics-server/templates/api-service.yaml (hunk 13 above)
Comment (ac74475) [reply charts/metrics-server/templates/api-service.yaml:28]: I am happy to merge this and if the discussion yields a need to change it then that can be another PR

File: Jenkinsfile (hunk 14 above)
Comment (ac74475) [reply Jenkinsfile]: reverted

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**